kustomization/components/samba-upload/deployment.yml

```diff
@@ -0,0 +1,54 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: samba
+spec:
+  revisionHistoryLimit: 4
+  strategy:
+    type: Recreate
```

Comment:
Do we need recreate here?  It is `ReadWriteMany`, and so therefore it can spin up the replacement deployment in parallel, I think, which will minimize downtime.